### ISSUE TITLE
Save TGV iterations

### ISSUE THREAD

**Issue by emilljungberg** [spinicist/riesling#1]
Option to save each iteration of the TGV recon could be useful for optimising number of iterations and for debugging. Would require some rewriting of the `tgv` function though.

Low priority.

**Comment by spinicist** [comment]
I have implemented such functionality for all the optimisers while writing them, and then removed it to clean up the code. It's straightforward to do a simple version of this, but there are a few tricky questions to generalise it:

1. Most of the optimisers maintain multiple variables, do we want to output all of them, or only the current image? Do we want a single true/false 'debug' flag or some kind of "level" that can determine what to write out?
2. Could this flag be integrated into the `Log` object rather than having to pass in another argument?
3. Do we need to write out the iteration images with correct voxel spacing/orientation? Then we have to pass in the info struct as well. If it's acceptable to write them out with the default 1mm iso spacing then we don't need to pass that in.
4. What to do with multi-volume/multi-echo images. The approach I took was to write the images out with filenames such as `tgv_u00.nii`, `tgv_u01.nii`, etc., but as currently each volume/echo is processed individually, only the last volume processed will persist.

We should probably take the pragmatic approach and add something simple for version 1 and come back later if it's insufficient.
